<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>BrushDock</class>
 <widget class="QWidget" name="BrushDock">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>325</width>
    <height>424</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="spacing">
    <number>3</number>
   </property>
   <property name="leftMargin">
    <number>3</number>
   </property>
   <property name="topMargin">
    <number>3</number>
   </property>
   <property name="rightMargin">
    <number>3</number>
   </property>
   <property name="bottomMargin">
    <number>3</number>
   </property>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <property name="spacing">
      <number>0</number>
     </property>
     <item>
      <widget class="QWidget" name="brushSlotContainer" native="true">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="GroupedToolButton" name="slot1">
          <property name="text">
           <string notr="true">1</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
          <property name="autoExclusive">
           <bool>true</bool>
          </property>
          <property name="groupPosition">
           <enum>GroupedToolButton::GroupLeft</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="GroupedToolButton" name="slot2">
          <property name="text">
           <string notr="true">2</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="autoExclusive">
           <bool>true</bool>
          </property>
          <property name="groupPosition">
           <enum>GroupedToolButton::GroupCenter</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="GroupedToolButton" name="slot3">
          <property name="text">
           <string notr="true">3</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="autoExclusive">
           <bool>true</bool>
          </property>
          <property name="groupPosition">
           <enum>GroupedToolButton::GroupCenter</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="GroupedToolButton" name="slot4">
          <property name="text">
           <string notr="true">4</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="autoExclusive">
           <bool>true</bool>
          </property>
          <property name="groupPosition">
           <enum>GroupedToolButton::GroupCenter</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="GroupedToolButton" name="slot5">
          <property name="text">
           <string notr="true">5</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="autoExclusive">
           <bool>true</bool>
          </property>
          <property name="groupPosition">
           <enum>GroupedToolButton::GroupCenter</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="GroupedToolButton" name="slotEraser">
          <property name="icon">
           <iconset theme="draw-eraser">
            <normaloff>theme:draw-eraser.svg</normaloff>theme:draw-eraser.svg</iconset>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="autoExclusive">
           <bool>true</bool>
          </property>
          <property name="groupPosition">
           <enum>GroupedToolButton::GroupRight</enum>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>5</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <widget class="BrushPreview" name="preview">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>70</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>128</height>
      </size>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_5">
     <item>
      <widget class="QWidget" name="brushTypeContainer" native="true">
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="GroupedToolButton" name="hardedgeMode">
          <property name="toolTip">
           <string>Pixel brush</string>
          </property>
          <property name="icon">
           <iconset>
            <normaloff>builtin:pixel-round.svg</normaloff>builtin:pixel-round.svg</iconset>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="autoExclusive">
           <bool>true</bool>
          </property>
          <property name="groupPosition">
           <enum>GroupedToolButton::GroupLeft</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="GroupedToolButton" name="squareMode">
          <property name="toolTip">
           <string>Square pixel brush</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="autoExclusive">
           <bool>true</bool>
          </property>
          <property name="groupPosition">
           <enum>GroupedToolButton::GroupCenter</enum>
          </property>
          <property name="icon">
           <iconset>
            <normaloff>builtin:square.svg</normaloff>builtin:square.svg</iconset>
          </property>
         </widget>
        </item>
        <item>
         <widget class="GroupedToolButton" name="softedgeMode">
          <property name="toolTip">
           <string>Brush</string>
          </property>
          <property name="icon">
           <iconset>
            <normaloff>builtin:round.svg</normaloff>builtin:round.svg</iconset>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="autoExclusive">
           <bool>true</bool>
          </property>
          <property name="groupPosition">
           <enum>GroupedToolButton::GroupCenter</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="GroupedToolButton" name="watercolorMode">
          <property name="toolTip">
           <string>Watercolor</string>
          </property>
          <property name="icon">
           <iconset theme="draw-watercolor">
            <normaloff>theme:draw-watercolor.svg</normaloff>theme:draw-watercolor.svg</iconset>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="autoExclusive">
           <bool>true</bool>
          </property>
          <property name="groupPosition">
           <enum>GroupedToolButton::GroupRight</enum>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="GroupedToolButton" name="modeEraser">
       <property name="toolTip">
        <string>Eraser mode</string>
       </property>
       <property name="icon">
        <iconset theme="draw-eraser">
         <normaloff>theme:draw-eraser.svg</normaloff>theme:draw-eraser.svg</iconset>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QComboBox" name="blendmode"/>
     </item>
     <item>
      <widget class="GroupedToolButton" name="modeIncremental">
       <property name="toolTip">
        <string>Direct drawing mode</string>
       </property>
       <property name="icon">
        <iconset>
         <normaloff>builtin:paintmode-incremental.svg</normaloff>builtin:paintmode-incremental.svg</iconset>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QGridLayout" name="gridLayout_2">
     <property name="horizontalSpacing">
      <number>3</number>
     </property>
     <property name="verticalSpacing">
      <number>0</number>
     </property>
     <item row="4" column="1">
      <widget class="QSlider" name="colorpickup">
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>32</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="4" column="2">
      <widget class="QSpinBox" name="colorpickupBox">
       <property name="prefix">
        <string notr="true">1/</string>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>32</number>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QSlider" name="brushsize">
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>128</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QSpinBox" name="spinBox">
       <property name="suffix">
        <string>px</string>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>128</number>
       </property>
      </widget>
     </item>
     <item row="2" column="2">
      <widget class="QSpinBox" name="hardnessBox">
       <property name="suffix">
        <string notr="true">%</string>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QSpinBox" name="spinBox_2">
       <property name="suffix">
        <string notr="true">%</string>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="hardnessLabel">
       <property name="text">
        <string>Hardness:</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QSlider" name="brushhardness">
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QSlider" name="brushopacity">
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Opacity:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Size:</string>
       </property>
      </widget>
     </item>
     <item row="5" column="0">
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>Spacing:</string>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="smudgingLabel">
       <property name="text">
        <string>Smudging:</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QSlider" name="brushsmudging">
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="3" column="2">
      <widget class="QSpinBox" name="smudgingBox">
       <property name="suffix">
        <string notr="true">%</string>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="colorpickupLabel">
       <property name="text">
        <string>Color pickup:</string>
       </property>
      </widget>
     </item>
     <item row="5" column="1">
      <widget class="QSlider" name="brushspacing">
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>150</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="5" column="2">
      <widget class="QSpinBox" name="spinBox_4">
       <property name="suffix">
        <string notr="true">%</string>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>150</number>
       </property>
      </widget>
     </item>
     <item row="6" column="0">
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>Pressure:</string>
       </property>
      </widget>
     </item>
     <item row="6" column="1" colspan="2">
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0">
        <widget class="QCheckBox" name="pressureSize">
         <property name="text">
          <string>Size</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1" colspan="2">
        <widget class="QCheckBox" name="pressureOpacity">
         <property name="text">
          <string>Opacity</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QCheckBox" name="pressureHardness">
         <property name="text">
          <string>Hardness</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1" colspan="2">
        <widget class="QCheckBox" name="pressureSmudging">
         <property name="text">
          <string>Smudging</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <spacer name="verticalSpacer_2">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>BrushPreview</class>
   <extends>QFrame</extends>
   <header>widgets/brushpreview.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>GroupedToolButton</class>
   <extends>QToolButton</extends>
   <header>widgets/groupedtoolbutton.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>brushsize</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>223</x>
     <y>189</y>
    </hint>
    <hint type="destinationlabel">
     <x>273</x>
     <y>182</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>brushsize</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>263</x>
     <y>190</y>
    </hint>
    <hint type="destinationlabel">
     <x>229</x>
     <y>192</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>brushopacity</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spinBox_2</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>194</x>
     <y>245</y>
    </hint>
    <hint type="destinationlabel">
     <x>304</x>
     <y>231</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_2</sender>
   <signal>valueChanged(int)</signal>
   <receiver>brushopacity</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>260</x>
     <y>241</y>
    </hint>
    <hint type="destinationlabel">
     <x>202</x>
     <y>244</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>brushhardness</sender>
   <signal>valueChanged(int)</signal>
   <receiver>hardnessBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>213</x>
     <y>293</y>
    </hint>
    <hint type="destinationlabel">
     <x>295</x>
     <y>281</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hardnessBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>brushhardness</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>290</x>
     <y>287</y>
    </hint>
    <hint type="destinationlabel">
     <x>219</x>
     <y>300</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>brushsmudging</sender>
   <signal>valueChanged(int)</signal>
   <receiver>smudgingBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>206</x>
     <y>347</y>
    </hint>
    <hint type="destinationlabel">
     <x>284</x>
     <y>331</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>smudgingBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>brushsmudging</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>253</x>
     <y>343</y>
    </hint>
    <hint type="destinationlabel">
     <x>222</x>
     <y>345</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>colorpickup</sender>
   <signal>valueChanged(int)</signal>
   <receiver>colorpickupBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>214</x>
     <y>369</y>
    </hint>
    <hint type="destinationlabel">
     <x>305</x>
     <y>372</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>colorpickupBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>colorpickup</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>278</x>
     <y>385</y>
    </hint>
    <hint type="destinationlabel">
     <x>222</x>
     <y>379</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>brushspacing</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spinBox_4</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>229</x>
     <y>409</y>
    </hint>
    <hint type="destinationlabel">
     <x>304</x>
     <y>409</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_4</sender>
   <signal>valueChanged(int)</signal>
   <receiver>brushspacing</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>279</x>
     <y>411</y>
    </hint>
    <hint type="destinationlabel">
     <x>221</x>
     <y>415</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
